server/test_commands.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/plugin"
+)
+
+const (
+	// ENABLE_TEST_COMMANDS can be set to true at compile time to enable some additional testing commands.
+	ENABLE_TEST_COMMANDS = false
+
+	COMMAND_TEST         = "nps-test"
+	COMMAND_TEST_RESET   = "reset"
+	COMMAND_TEST_VERSION = "version"
+)
+
+func (p *Plugin) registerCommands() {
+	if ENABLE_TEST_COMMANDS {
+		p.API.RegisterCommand(&model.Command{
+			Trigger: COMMAND_TEST,
+		})
+	}
+}
+
+func parseCommandArgs(commandArgs *model.CommandArgs) (string, string, []string) {
+	split := strings.Fields(commandArgs.Command)
+
+	command := ""
+	if len(split) > 0 {
+		command = split[0]
+
+		// Trim the leading slash
+		command = command[1:]
+	}
+
+	action := ""
+	if len(split) > 1 {
+		action = split[1]
+	}
+
+	var args []string
+	if len(split) > 2 {
+		args = split[2:]
+	}
+
+	return command, action, args
+}
+
+func (p *Plugin) ExecuteCommand(c *plugin.Context, commandArgs *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	command, action, args := parseCommandArgs(commandArgs)
+
+	if ENABLE_TEST_COMMANDS && command == COMMAND_TEST {
+		return p.executeTestCommand(action, args)
+	}
+
+	return nil, nil
+}
+
+func (p *Plugin) executeTestCommand(action string, args []string) (*model.CommandResponse, *model.AppError) {
+	if action == "" {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         fmt.Sprintf("Unknown action %s specified.", action),
```

Comment:
The error message is weird, actually you give here the "action" name information when the action is empty, but not some lines bellow where the action is invalid and not empty.

Reply:
Oops. When addressing Ben's concerns, I accidentally replaced the wrong error message. I'll fix that